src/parquet/arrow/arrow-schema-test.cc

```diff
@@ -681,6 +682,65 @@ TEST_F(TestConvertArrowSchema, ParquetFlatPrimitives) {
   CheckFlatSchema(parquet_fields);
 }
 
+TEST_F(TestConvertArrowSchema, ParquetFlatPrimitivesAsDictionaries) {
+  std::vector<NodePtr> parquet_fields;
+  std::vector<std::shared_ptr<Field>> arrow_fields;
+  std::shared_ptr<::arrow::Array> dict;
+
+  parquet_fields.push_back(
+      PrimitiveNode::Make("int32", Repetition::REQUIRED, ParquetType::INT32));
+  ArrayFromVector<::arrow::Int32Type, int32_t>(std::vector<int32_t>(), &dict);
+  arrow_fields.push_back(::arrow::field("int32", ::arrow::dictionary(::arrow::int8(), dict), false));
+
+  parquet_fields.push_back(
+      PrimitiveNode::Make("int64", Repetition::REQUIRED, ParquetType::INT64));
+  ArrayFromVector<::arrow::Int64Type, int64_t>(std::vector<int64_t>(), &dict);
+  arrow_fields.push_back(std::make_shared<Field>("int64", ::arrow::dictionary(::arrow::int8(), dict), false));
+
+  parquet_fields.push_back(PrimitiveNode::Make("date", Repetition::REQUIRED,
+                                               ParquetType::INT32, LogicalType::DATE));
+  ArrayFromVector<::arrow::Date32Type, int32_t>(std::vector<int32_t>(), &dict);
+  arrow_fields.push_back(std::make_shared<Field>("date", ::arrow::dictionary(::arrow::int8(), dict), false));
+
+  parquet_fields.push_back(PrimitiveNode::Make("date64", Repetition::REQUIRED,
+                                               ParquetType::INT32, LogicalType::DATE));
+  ArrayFromVector<::arrow::Date64Type, int64_t>(std::vector<int64_t>(), &dict);
+  arrow_fields.push_back(std::make_shared<Field>("date64", ::arrow::dictionary(::arrow::int8(), dict), false));
+
+  /*parquet_fields.push_back(PrimitiveNode::Make("timestamp", Repetition::REQUIRED,
```

Comment:
These comments need to implemented otherwise this is good to go when https://github.com/apache/arrow/pull/1086 is merged.